### PR TITLE
Improve: cleanup OSConfig and refacto CLI's tests to run outside real QGIS profiles folder

### PR DESCRIPTION
**Pull request by Guts**
Some fixes and clean up following https://github.com/Guts/qgis-deployment-cli/pull/421

### CODE REVIEW

File: qgis_deployment_toolbelt/constants.py [hunk 1]
```diff
@@ -225,7 +228,7 @@ def valid_shortcut_name(self, shortcut_name: str) -> bool:
     @classmethod
     def from_opersys(
         cls,
-        operating_system_codename: Literal["darwin", "linux", "win32"] | None = None,
+        operating_system_codename: str | None = None,
```
Comment (Ducarouge) [inline qgis_deployment_toolbelt/constants.py:231]: Why `opersys` is not set as default value ? Just for the log ?

File: tests/test_cli_main.py [hunk 2]
```diff
@@ -79,13 +83,39 @@ def test_cli_version(capsys, option):
...
-    assert err == ""
+    with tempfile.TemporaryDirectory(
+        prefix="QDT_test_cli_run_",
+        ignore_cleanup_errors=True,
+    ) as tmpdirname:
+        # customize QDT working folder and profiles destination folder
+        tmp_dir = Path(tmpdirname).joinpath(f"scenario_{good_scenarios.index(option)}")
+        environ[
+            "QDT_LOCAL_WORK_DIR"
+        ] = f"{tmp_dir.parent.joinpath('qdt_working_folder').resolve()}"
```
Comment (Ducarouge) [inline tests/test_cli_main.py:97]: `Path(tmpdirname).joinpath('qdt_working_folder').resolve()` would be easier to read.

File: tests/test_cli_main.py [hunk 3]
```diff
@@ -79,13 +83,39 @@ def test_cli_version(capsys, option):
 
 
 @pytest.mark.parametrize("option", good_scenarios)
-def test_main_run(capsys, option):
+def test_main_run(capsys, option, monkeypatch):
```
Comment (Ducarouge) [inline tests/test_cli_main.py:86]: Is `monkeypatch` used ?

File: tests/test_constants.py [hunk 4]
```diff
@@ -43,8 +42,8 @@ def setUp(self) -> None:
 
     def test_constants(self):
         """Test types."""
-        self.assertIsInstance(OS_CONFIG, dict)
-        os_config = OS_CONFIG.get(opersys)
+
+        os_config = OSConfiguration.from_opersys(opersys)
```
Comment (Ducarouge) [inline tests/test_constants.py:46]: argument `opersys` is already the implicit default value

File: tests/test_cli_main.py (hunk 3 above)
Comment (Guts) [reply tests/test_cli_main.py:86]: Removed.

File: qgis_deployment_toolbelt/constants.py (hunk 1 above)
Comment (Guts) [reply qgis_deployment_toolbelt/constants.py:231]: To anticipate over future operating system whom code name is not within opersys module.

File: tests/test_cli_main.py (hunk 2 above)
Comment (Guts) [reply tests/test_cli_main.py:97]: If you say that, why not. It was a way to have only one variable to manage (the second being defined by the first). Changed.

File: tests/test_constants.py (hunk 4 above)
Comment (Guts) [reply tests/test_constants.py:46]: You're right. Removed.

**Review by Ducarouge — APPROVED** [review]

**Pull request merged**